### DISCUSSION TITLE
VisualStudio now suggest to use `[field: ...]` instead of `[property: ...]`. Is it safe to switch?

### DISCUSSION THREAD

**Discussion by AndreasReitberger** [CommunityToolkit/dotnet#708] [Q&A]
As I started to use this toolkit, I needed to use the `[property: ...]` to pass attributes to the SourceGenerator.
Is this still so? Visual Studio says now to switch to field, however I wasn't able to find a change in the changelog indicating this.

![image](https://github.com/CommunityToolkit/dotnet/assets/2952443/e8775cb8-8fbe-4c6d-9c44-a57960e85124)

```cs
        // This?
        [ObservableProperty]
        [property: JsonIgnore, XmlIgnore]
        RepetierPrinterState state;
        partial void OnStateChanged(RepetierPrinterState value)
        {
            OnRepetierPrinterStateChangedEvent(new RepetierPrinterStateChangedEventArgs()
            {
                NewPrinterState = value,
                SessonId = SessionId,
                CallbackId = -1,
                Printer = GetActivePrinterSlug(),
            });
            UpdatePrinterState(value);
        }
       // Or this?
        [ObservableProperty]
        [field: JsonIgnore, XmlIgnore]
        RepetierCurrentPrintInfo activePrintInfo;
        partial void OnActivePrintInfoChanged(RepetierCurrentPrintInfo value)
        {
            OnPrintInfoChangedEvent(new RepetierActivePrintInfoChangedEventArgs()
            {
                SessonId = SessionId,
                NewActivePrintInfo = value,
                Printer = GetActivePrinterSlug(),
            });
            UpdateActivePrintInfo(value);
        }
```

**Comment by Sergio0694** [comment]
That is still the case, yes. If you want to pass an attribute to the generated property, `property:` is the correct thing to use. It seems the diagnostic suppressor isn't working as expected here. Are you able to create a minimal repro (possibly in a standalone project) that reproduces this, so we can investigate it? Also have you tried restarting your IDE just in case?

**Reply by AndreasReitberger** [reply]
Thanks for the clarification. I used the current Preview version of Visual Studio. There, I get this warning. If I open it with the stable version of Visual Studio, it seems to work. 

![image](https://github.com/CommunityToolkit/dotnet/assets/2952443/9a53f9c4-aca1-4159-9e31-9e28fd8f4aae)

So I guess it's not a problem of the project it self, but from the Visual Studio version.

Gives warnings
![image](https://github.com/CommunityToolkit/dotnet/assets/2952443/c5a22394-cd1c-450a-9e5b-3173065f47cd)

Doesn't warn
![image](https://github.com/CommunityToolkit/dotnet/assets/2952443/051599ed-e75d-4783-8a8b-33dc1f320dc3)

**Reply by Sergio0694** [reply]
Would you mind opening an issue about this in the [Roslyn repo](https://github.com/dotnet/roslyn), just for tracking? Given that it'd seem to be a regression. Feel free to link back to this discussion, and possibly provide a minimal repro they can also use. Thank you! 🙂
